graph/denota/src/nota-registrar.ts

```diff
@@ -0,0 +1,242 @@
+import {
+  Approval as ApprovalEvent,
+  ApprovalForAll as ApprovalForAllEvent,
+  BatchMetadataUpdate as BatchMetadataUpdateEvent,
+  Cashed as CashedEvent,
+  ContractURIUpdated as ContractURIUpdatedEvent,
+  Funded as FundedEvent,
+  MetadataUpdate as MetadataUpdateEvent,
+  ModuleWhitelisted as ModuleWhitelistedEvent,
+  OwnershipTransferred as OwnershipTransferredEvent,
+  TokenWhitelisted as TokenWhitelistedEvent,
+  Transfer as TransferEvent,
+  Transferred as TransferredEvent,
+  Written as WrittenEvent
+} from "../generated/NotaRegistrar/NotaRegistrar"
+import {
+  Approval,
+  ApprovalForAll,
+  BatchMetadataUpdate,
+  Cashed,
+  ContractURIUpdated,
+  Funded,
+  MetadataUpdate,
+  ModuleWhitelisted,
+  OwnershipTransferred,
+  TokenWhitelisted,
+  Transfer,
+  Transferred,
+  Written
+} from "../generated/schema"
+
+export function handleApproval(event: ApprovalEvent): void {
+  let entity = new Approval(
+    event.transaction.hash.concatI32(event.logIndex.toI32())
+  )
+  entity.owner = event.params.owner
+  entity.approved = event.params.approved
+  entity.tokenId = event.params.tokenId
+
+  entity.blockNumber = event.block.number
+  entity.blockTimestamp = event.block.timestamp
+  entity.transactionHash = event.transaction.hash
+
+  entity.save()
+}
+
+export function handleApprovalForAll(event: ApprovalForAllEvent): void {
+  let entity = new ApprovalForAll(
+    event.transaction.hash.concatI32(event.logIndex.toI32())
+  )
+  entity.owner = event.params.owner
+  entity.operator = event.params.operator
+  entity.approved = event.params.approved
+
+  entity.blockNumber = event.block.number
+  entity.blockTimestamp = event.block.timestamp
+  entity.transactionHash = event.transaction.hash
+
+  entity.save()
+}
+
+export function handleBatchMetadataUpdate(
+  event: BatchMetadataUpdateEvent
+): void {
+  let entity = new BatchMetadataUpdate(
+    event.transaction.hash.concatI32(event.logIndex.toI32())
+  )
+  entity._fromTokenId = event.params._fromTokenId
+  entity._toTokenId = event.params._toTokenId
+
+  entity.blockNumber = event.block.number
+  entity.blockTimestamp = event.block.timestamp
+  entity.transactionHash = event.transaction.hash
+
+  entity.save()
+}
+
+export function handleCashed(event: CashedEvent): void {
+  let entity = new Cashed(
```

Comment:
Is it not possible to update entities at all? Ideally we would have a "Nota" entity for each nota that gets updated with the current state. Otherwise we have to put this information together in the frontend

Reply:
The Nota entities should be created and other ones deleted if they won't be used by the frontend. These were auto-generated so served as a good starting point